backend/src/models/tag.ts

```diff
@@ -1,53 +1,60 @@
 import { Schema, Types, model } from "mongoose";
 
 export interface ITag {
-	_id: Types.ObjectId;
-	name: string;
-	tagColor: string;
-	slug: string;
-	user: Types.ObjectId;
-	workspace: Types.ObjectId;
+  _id: Types.ObjectId;
+  name: string;
+  tagColor: string;
+  slug: string;
+  user: Types.ObjectId;
+  workspace: Types.ObjectId;
+  checkedSecrets?: { _id: string; isChecked: string | boolean }[];
 }
-
 const tagSchema = new Schema<ITag>(
-	{
-		name: {
-			type: String,
-			required: true,
-			trim: true,
-		},
-		tagColor: {
-			type: String,
-			required: false,
-			trim: true,
-		},
-		slug: {
-			type: String,
-			required: true,
-			trim: true,
-			lowercase: true,
-			validate: [
-				function (value: any) {
-					return value.indexOf(" ") === -1;
-				},
-				"slug cannot contain spaces",
-			],
-		},
-		user: {
-			type: Schema.Types.ObjectId,
-			ref: "User",
-		},
-		workspace: {
-			type: Schema.Types.ObjectId,
-			ref: "Workspace",
-		},
-	},
-	{
-		timestamps: true,
-	}
+  {
+    name: {
+      type: String,
+      required: true,
+      trim: true
+    },
+    tagColor: {
+      type: String,
+      required: false,
+      trim: true
+    },
+    slug: {
+      type: String,
+      required: true,
+      trim: true,
+      lowercase: true,
+      validate: [
+        function (value: string) {
+          return value.indexOf(" ") === -1;
+        },
+        "slug cannot contain spaces"
+      ]
+    },
+    user: {
+      type: Schema.Types.ObjectId,
+      ref: "User"
+    },
+    workspace: {
+      type: Schema.Types.ObjectId,
+      ref: "Workspace"
+    },
+    checkedSecrets: {
```

Comment:
Can you tell me the purpose of this. I don't think we need a two way relationship between tag and secret

Reply:
I am now checking if folder exist before moving secrets @akhilmhdh

Reply:
![image](https://github.com/Infisical/infisical/assets/23234918/32e0148b-f61e-4a99-9a5d-4a7af588c0cc)

Here is what  pushed earlier to check if folder exist and if it workspace and environment are valid. As well as putting restriction to the operation. In case you didint see the code

Reply:
@EBEN4REAL i did saw that. And that's what i requested. Here your fetching the folder list of an workspace and its environment. Your not checking the provided folder id is present inside that folder list. Kindly refer the folder service functions to fetch folder from folder list

Reply:
And regarding the checked secrets what i was asking for putting it in wsTags list in frontend. We don't want another relation towards like this

Reply:
Okay, got it!

Reply:
I have added a fix for this also. kindly check and let me know what you think.